src/pathwise_sampling.jl

```diff
@@ -0,0 +1,94 @@
+@doc raw"""
+    pathwise_sample(rng::Random.AbstractRNG, f::ApproxPosteriorGP, weight_space_approx[, num_samples::Integer])
+
+Efficiently samples a function from a sparse approximate posterior GP `f`.
+Returns a function which can be evaluated at any input locations `X`.
+`weight_space_approx` must be a function which takes a prior `AbstractGP` as an
+argument and returns a `BayesianLinearRegressors.BasisFunctionRegressor`,
+representing a weight space approximation to the prior of `f`. An example of
+such a function can be constructed with
+`RandomFourierFeatures.build_rff_weight_space_approx`.
+
+If `num_samples` is supplied as an argument, returns a Vector of function
+samples.
+
+Details of the method can be found in [1].
+
+[1] - Wilson, James, et al. "Efficiently sampling functions from Gaussian
+process posteriors." International Conference on Machine Learning. PMLR, 2020.
+"""
+function pathwise_sample(rng::Random.AbstractRNG, f::ApproxPosteriorGP, weight_space_approx)
+    prior_approx = weight_space_approx(f.prior)
+    prior_sample = rand(rng, prior_approx)
+
+    z = inducing_points(f)
+    q_u = _get_q_u(f)
+
+    u = rand(rng, q_u)
+    v = cov(f, z) \ (u - prior_sample(z))
+
+    posterior_sample(x) = prior_sample(x) + cov(f, x, z) * v
```

Comment:
I imagine it's better to use a callable struct instead of a closure here - but it looks so pretty...